File: examples/governance_static_go/domain/set_threshold.go
package domain

import (
	"time"
	"fmt"
)

type SetThreshold struct {
	MonitoringId string `json:"monitoring_id"`
	Threshold float64 `json:"threshold"`
}

func (c SetThreshold) CommandName() string { return "SetThreshold" }

func (c SetThreshold) Execute(repo MonitoringRepository) (*Monitoring, *SetThresholdEvent, error) {
	existing, err := repo.Find(c.MonitoringId)
	if err != nil {
		return nil, nil, err
	}
	if existing == nil {
		return nil, nil, fmt.Errorf("Monitoring not found: %s", c.MonitoringId)
	}
	existing.Threshold = c.Threshold
	existing.UpdatedAt = time.Now()
	if err := existing.Validate(); err != nil {
		return nil, nil, err
	}
	if err := repo.Save(existing); err != nil {
		return nil, nil, err
	}
	event := SetThresholdEvent{
		AggregateID: existing.ID,
		MonitoringId: c.MonitoringId,
		Threshold: c.Threshold,
		OccurredAt: time.Now(),
	}
	return existing, &event, nil
}

File: examples/governance/operations_domain/lib/operations_domain/monitoring/events/set_threshold.rb
module OperationsDomain
  class Monitoring
    module Events
      class SetThreshold
        attr_reader :aggregate_id, :monitoring_id, :threshold, :model_id, :deployment_id, :metric_name, :value, :recorded_at, :occurred_at

        def initialize(aggregate_id: nil, monitoring_id: nil, threshold: nil, model_id: nil, deployment_id: nil, metric_name: nil, value: nil, recorded_at: nil)
          @aggregate_id = aggregate_id
          @monitoring_id = monitoring_id
          @threshold = threshold
          @model_id = model_id
          @deployment_id = deployment_id
          @metric_name = metric_name
          @value = value
          @recorded_at = recorded_at
          @occurred_at = Time.now
          freeze
        end
      end
    end
  end
end
